ocpp/charge_point.py

```diff
@@ -290,15 +303,18 @@ async def call(self, payload, suppress=True):
             raise response.to_exception()
         else:
             response.action = call.action
-            validate_payload(response, self._ocpp_version)
+            validate_payload(response, self._ocpp_version, extension=extension)
 
         snake_case_payload = camel_to_snake_case(response.payload)
         # Create the correct Payload instance based on the received payload. If
         # this method is called with a call.BootNotificationPayload, then it
         # will create a call_result.BootNotificationPayload. If this method is
         # called with a call.HeartbeatPayload, then it will create a
         # call_result.HeartbeatPayload etc.
-        cls = getattr(self._call_result, payload.__class__.__name__)  # noqa
+        if extension:
+            cls = getattr(extension_call_result, payload.__class__.__name__)  # noqa
+        else:
+            cls = getattr(self._call_result, payload.__class__.__name__)  # noqa
```

Comment:
@OrangeTux I feel there is probably a better way to handle this situation, but this is where I landed. I am open to suggestions to improve this.